### PR TITLE
Add Cmake and Github CI

### LINKED ISSUE #21: Migrate to CMake
I tried to build this project with FetchContent in cmake
```cmake
cmake_minimum_required(VERSION 3.11)

project (uvwasi)

include(FetchContent)

if (APPLE)
   set(CMAKE_MACOSX_RPATH ON)
endif (APPLE)

## https://libuv.org
FetchContent_Declare(
        libuv
        GIT_REPOSITORY https://github.com/libuv/libuv.git
        GIT_TAG v1.32.0)

FetchContent_GetProperties(libuv)
if(NOT libuv_POPULATED)
    FetchContent_Populate(libuv)
    include_directories(
            "${libuv_SOURCE_DIR}/include"
    )
    add_subdirectory(
            ${libuv_SOURCE_DIR}
            ${libuv_BINARY_DIR}
            EXCLUDE_FROM_ALL)
endif()

set(SOURCES
    app.c
    src/fd_table.c
    src/uv_mapping.c
    src/uvwasi.c
)

add_executable(uvwasi ${SOURCES})

target_include_directories(${PROJECT_NAME}
    PRIVATE
    ${PROJECT_SOURCE_DIR}/include
)

#target_link_libraries(${PROJECT_NAME}
#    PRIVATE
#    libuv
#)

```

failed with 
```console
Scanning dependencies of target uvwasi
[ 50%] Building C object CMakeFiles/uvwasi.dir/app.c.o
[ 50%] Building C object CMakeFiles/uvwasi.dir/src/uvwasi.c.o
[ 50%] Building C object CMakeFiles/uvwasi.dir/src/uv_mapping.c.o
[ 50%] Building C object CMakeFiles/uvwasi.dir/src/fd_table.c.o
[100%] Linking C executable uvwasi
Undefined symbols for architecture x86_64:
  "_uv_buf_init", referenced from:
```

cc @danbev @saghul

### PR DESCRIPTION
**Pull request by gengjiawen**
Fix #21 

Great thanks to @saghul for instruct me on Cmake.

Build works on ubuntu and macOS now: see https://github.com/gengjiawen/uvwasi/runs/255150824. 
Windows looks like false positive.

### CODE REVIEW

**Comment by danbev** [comment]
I think we need to update `app.c` to use a different path for [foo.txt](https://github.com/cjihrig/uvwasi/blob/29ae590ccd2f35879b429f2ec3ef6d9691e6e953/app.c#L76). Perhaps it should now be:
```console
diff --git a/app.c b/app.c
index 3470583..322e550 100644
--- a/app.c
+++ b/app.c
@@ -73,7 +73,7 @@ int main(void) {

   uvwasi_fd_t dirfd = 3;
   uvwasi_lookupflags_t dirflags = 1;
-  const char* path = "../uvwasi/./foo.txt";
+  const char* path = "../build/./foo.txt";
   uvwasi_oflags_t o_flags = UVWASI_O_CREAT;
   uvwasi_rights_t fs_rights_base = UVWASI_RIGHT_FD_DATASYNC |
                                    UVWASI_RIGHT_FD_ALLOCATE |
```
Without this, I get the following error when trying to run the executable:
```console
$ mkdir build
$ cd build
$ cmake ..
$ cmake --build .
[ 81%] Built target uv_a
Scanning dependencies of target uvwasi
[ 81%] Building C object CMakeFiles/uvwasi.dir/app.c.o
[ 90%] Linking C executable uvwasi
[100%] Built target uvwasi
$ ./uvwasi
Assertion failed: (r == 0), function main, file /Users/danielbevenius/work/wasm/uvwasi/app.c, line 102.
Abort trap: 6
```

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+      - name: build it
+        run: |
+          npx envinfo
+          cmake .
```
Comment (gengjiawen) [inline .github/workflows/nodejs.yml]: @danbev Should I change this to:
```bash
mkdir build
cd build
cmake ..
cmake --build .
```

Comment (gengjiawen) [reply .github/workflows/nodejs.yml]: Also do I need to delete `Makefile`

Comment (danbev) [reply .github/workflows/nodejs.yml]: I don't think it matters on the CI system (having the generated files mixed with the sources is more a preference when running locally I think.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Node CI
```
Comment (cjihrig) [inline .github/workflows/nodejs.yml]: This file is named `nodejs.yml`, and there are a few other mentions of Node throughout this file. Is that just leftover from copy/paste? If so, can you remove them? I'm really hoping that we can end up landing this in Node, but it should also be able to run anywhere else that libuv can.

**Comment by cjihrig** [comment]
Thank you so much for taking this on @gengjiawen 😄

File: .github/workflows/nodejs.yml (hunk 2 above)
Comment (gengjiawen) [reply .github/workflows/nodejs.yml]: It's copied from one of my Node.js project, I will try to change it :)

File: app.c [hunk 3]
```diff
@@ -73,7 +73,7 @@ int main(void) {
 
   uvwasi_fd_t dirfd = 3;
   uvwasi_lookupflags_t dirflags = 1;
-  const char* path = "../uvwasi/./foo.txt";
+  const char* path = "../build/./foo.txt";
```
Comment (gengjiawen) [inline app.c]: Can this change to a tempdir ?

Comment (gengjiawen) [reply app.c]: Or just `const char* path = "./foo.txt";`

**Pull request merged**